manual/src/tutorials/moduleexamples.etex

```diff
@@ -130,50 +114,54 @@ Signatures are interfaces for structures. A signature specifies
 which components of a structure are accessible from the outside, and
 with which type. It can be used to hide some components of a structure
 (e.g. local function definitions) or export some components with a
-restricted type. For instance, the signature below specifies the three
-priority queue operations "empty", "insert" and "extract", but not the
-auxiliary function "remove_top". Similarly, it makes the "queue" type
+restricted type. For instance, the signature below specifies the
+queue operations "empty", "add", "top" and "pop", but not the
+auxiliary function "make". Similarly, it makes the "queue" type
 abstract (by not providing its actual representation as a concrete type).
+This ensures that users of the "Fifo" module cannot violate data
+structure invariants that operations rely on, such as ``if the front list
+is empty, the rear list is empty too''.
 \begin{caml_example}{toplevel}
-module type PRIOQUEUE =
+module type FIFO =
   sig
-    type priority = int         (* still concrete *)
-    type 'a queue               (* now abstract *)
+    type 'a queue               (* now an abstract type *)
     val empty : 'a queue
-    val insert : 'a queue -> int -> 'a -> 'a queue
-    val extract : 'a queue -> int * 'a * 'a queue
-    exception Queue_is_empty
+    val add : 'a -> 'a queue -> 'a queue
+    val top : 'a queue -> 'a
+    val pop : 'a queue -> 'a queue
+    exception Empty
   end;;
 \end{caml_example}
-Restricting the "PrioQueue" structure by this signature results in
-another view of the "PrioQueue" structure where the "remove_top"
-function is not accessible and the actual representation of priority
+Restricting the "Fifo" structure by this signature results in
```

Comment:
"restrict by" or "restrict to"? The second sounds better to me. 
```suggestion
Restricting the "Fifo" structure to this signature results in
```

Reply:
OK for "restrict to".